### PR TITLE
use /dev/disk/by-path for better nvme support

### PR DESCRIPTION
**Pull request by Lassulus**

### CODE REVIEW

File: lib/default.nix [hunk 1]
```diff
@@ -9,6 +9,25 @@ let {
   body.mount = mount-f {};
 
 
+  helper.find-device = device: let
+    environment = helper.device-id device;
+  in ''
+    if echo '${device} | grep -q '^/dev/disk'; then
```
Comment (zimbatm) [inline lib/default.nix:15]:
```suggestion
    if echo '${device}' | grep -q '^/dev/disk'; then
```
BUG!

**Comment by Baitinq** [comment]
@Lassulus Thank you for your amazing work.

**Comment by Mic92** [comment]
> @Lassulus Thank you for your amazing work.
> 
> EDIT: What do you think about encouraging this new way of identifying disks? Imho I think its much better as it is more deterministic.

We should document both versions but there are also good reasons to use non-deterministic approaches i.e. when provisioning servers where you don't know the ids in advance.

File: tests/test.nix [hunk 2]
```diff
@@ -107,6 +107,7 @@ in makeTest' {
 
   testScript = ''
     machine.succeed("echo 'secret' > /tmp/secret.key");
+    machine.succeed("ls -la /dev/disk/by-id");
```
Comment (Mic92) [inline tests/test.nix:110]:
```suggestion
```

**Comment by Baitinq** [comment]
@Lassulus Why did you end up choosing by-path instead of by-id? Just curious, thank you for your work again.

**Comment by Lassulus** [comment]
@Baitinq because of https://github.com/nix-community/disko/pull/15#issuecomment-1223661848 :)

**Pull request merged**